api/src/services/assets.ts

```diff
@@ -86,16 +105,32 @@ export class AssetsService {
 		 */
 		if (!isValidUuid(id)) throw new ForbiddenError();
 
+		let allowedFields: string[] | undefined;
+
 		if (systemPublicKeys.includes(id) === false && this.accountability) {
-			await validateAccess(
-				{
-					accountability: this.accountability,
-					action: 'read',
-					collection: 'directus_files',
-					primaryKeys: [id],
-				},
-				{ knex: this.knex, schema: this.schema },
-			);
+			if (this.accountability.admin) {
+				allowedFields = ['*'];
+			} else {
```

Comment:
This needs to be adjusted as `this.accountability === null` is also admin. That will never be the case here due to the check above.

As `sanitizeFields` accepts undefined for allowedFields perhaps only assigning if they are not adming would be more appropriate
```suggestion
      let allowedFields;
			if (req.accountability?.admin !== true) {
```